### PR TITLE
[salt-ssh] Add a range roster and range targeting options for the flat roster

### PR DESCRIPTION
**Pull request by thomasdesr**
This PR has two changes in it (let me know if you want them split):
- Add support for targeting using a range query to the flat roster system
- Add a roster that uses range data as its backend. The major caveat here is that it currently has no additional support for any other targeting methods :S. However I think most folks who use range tend to use it as their central metadata store so this should get them the basic level of support they need.

### CODE REVIEW

**Comment by cachedout** [comment]
@thomaso-mirodin Nice! This will be very useful. 

There are a couple of minor lint errors here. Would you mind taking a look and fixing them before we get this merged? Thanks!

**Comment by thomaso-mirodin** [comment]
@cachedout I don't really know what to make of the last linting error, it seems to be concerned about a blank line XD

File: salt/roster/range.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+:depends: seco.range, https://github.com/ytoolshed/range
+
+When you want to use a range query for target matching, use ``--roster range``. For example:
+
+.. code-block:: bash
+
+    salt-ssh --roster range '%%%example.range.cluster' test.ping
+
+'''
+
```
Comment (cachedout) [inline salt/roster/range.py]: Please add `from __future__ import absolute_import`

**Comment by cachedout** [comment]
@thomaso-mirodin Yeah, that's a bug in the way the linter displays its output in Jenkins. Sorry about that. I've commented in-line with the line that needs to be added to satisfy the linter. Thanks!

**Comment by thomaso-mirodin** [comment]
@cachedout Looks like the the last failures are unrelated to salt-ssh, see anything else you'd like me to change? :D

**Comment by cachedout** [comment]
@thomaso-mirodin I think we have everything we need. I'm going to go ahead and merge this. Thanks!

**Comment by thomaso-mirodin** [comment]
@cachedout Hey! :D

Why the switch to using format on the log messages? That causes the cost of having to format strings to be paid regardless of if that logging level is enabled or not.

https://docs.python.org/2/library/logging.html#logging.Logger.debug

**Pull request merged**